nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/convolution/DepthwiseConv2D.java

```diff
@@ -308,7 +313,7 @@ public String opName() {
 
     @Override
     public List<SDVariable> doDiff(List<SDVariable> f1) {
-        throw new UnsupportedOperationException("Not implemented yet");
+        return Arrays.asList(new DepthwiseConv2DBp(sameDiff, f1, this.config).outputVariables());
```

Comment:
We discussed this.
```
in general (for pretty much every op)
every floating point output of forward pass is a gradient input of backprop
every floating point input of forward pass is a gradient output of backprop
i.e., if X = {x0, x1, ..., xN} is inputs to forward pass
and Y = {y0, y1, ..., yM} is output of forward pass
i.e., Y = op(X)
then backprop is:
dX = opBp(X, dY)
```

You're doing this: `dX = opBp(dY)` not `dX = opBp(X, dY)`

Put another way: It doesn't match the C++ op (i.e., INPUT_VARIABLE(x) part)
https://github.com/eclipse/deeplearning4j/blob/master/libnd4j/include/ops/declarable/generic/nn/convo/depthwiseConv2d.cpp#L152-L155
the C++ op expects 3 or 4 inputs:
input, weights, gradO (i.e., arg(0), arg(1), f1.get(0)), OR
input, weights, bias, gradO (i.e., arg(0), arg(1), arg(2), f1.get(0))